### DISCUSSION TITLE
Multi-year DUPV profiles

### DISCUSSION THREAD

**Discussion by gschivley** [NREL/ReEDS-2.0#180] [Q&A: Data] [answered]
I'm trying to compile the new multiple weather year data from ReEDS and am looking for some help to understand the distributed PV capacity and profiles. I've previously taken state-level capacity and generation profiles directly from the [Cambium results](https://scenarioviewer.nrel.gov), which is based on the 2012 weather year. 

ReEDS now seems to have a file (`dupv_ba.h5` located at [/inputs/variability/multi_year](https://github.com/NREL/ReEDS-2.0/tree/main/inputs/variability/multi_year) with 7 weather years of capacity factors by BA and resource class. I've also found the files `DUPV_supply_curves_capacity_2018.csv` and `DUPV_supply_curves_capacity_NARIS.csv`, which seems to have available capacity by resource class within each BA. Finally, I see that the folder [dGen_Model_Inputs](https://github.com/NREL/ReEDS-2.0/tree/main/inputs/dGen_Model_Inputs) has various `distPVcap_*` files, the newest of which is Standard Scenarios 2022.

I could use distributed PV capacity from the files in ReEDS or directly from the [2023 Standard Scenarios results](https://scenarioviewer.nrel.gov/?project=03ad535d-f0ed-4a6b-8323-c582a606e92c&mode=download&layout=Default), but both give total capacity per BA and the generation capacity factors are by resource class within BA. I haven't found anything in the ReEDS code that deals with this. Should I fill up resources class capacity starting with the lowest number and create a weighted average capacity factor profile? Or is there another set of data that I should look at?

**Comment by wesleyjcole** [comment, marked as answer]
ReEDS has two distributed PV capacity inputs: DUPV and distPV.  DUPV is distributed utility PV in urban region, and distPV is behind-the-meter rooftop PV.  We disabled DUPV a while ago (maybe about a year ago?) because we merged the DUPV with the UPV, so now all utility PV is captured in the UPV supply curves and generation profiles.  The distPV hourly profiles come from dGen, which only has a single year of weather data.  Therefore, to approximate distPV profiles in ReEDS, we use the UPV profiles, but scale them to match the annual capacity factors produced by dGen.  See https://github.com/NREL/ReEDS-2.0/blob/main/input_processing/LDC_prep.py#L156 for where that happens in ReEDS.

**Reply by gschivley** [reply]
Wes, can you confirm that the `distPVCF_hourly*.csv` files are all in local time? Looks like I can use `reeds_region_tz_map.csv` to shift them to a reference timezone.

While I'm asking, are all time series data stored in local time? Looks like the new EER demand (`EER_IRAmoderate_load_hourly.h5`) are probably also local.

**Reply by wesleyjcole** [reply]
Yes, these are in local time.  We convert to eastern time in the input processing scripts when a run gets kicked off.  At some point we'll change the inputs to all be in the same time zone rather than local time, so just a heads up that things could change down the road.

**Reply by gschivley** [reply]
Are the UPV profiles that are used as the basis of distPV aggregated within BA or for individual sites?

**Reply by wesleyjcole** [reply]
We use the worst UPV profile within a BA for the distPV profiles.  See https://github.com/NREL/ReEDS-2.0/blob/9b159a182a95bd7ff96bb8fa2f9d1110744fb51a/input_processing/LDC_prep.py#L156.

**Comment by gschivley** [comment]
Thanks for the quick answer, and good to know about the DUPV/distPV difference. Additional data files have been added to ReEDS over time and it can be difficult to determine which are current vs legacy.

The code in `get_distpv_profiles` doesn't seem to account for differences in distPV profiles by year. Is that correct? I've noticed that Cambium 2022 has different capacity factors for each model year. They generally drop over time, which I assumed was due to installing panels on better roof orientations in early years. But I just looked at a selection of normalized profiles for Iowa and am not sure if that's what is happening. Is distributed PV being curtailed in the Cambium results?

![image](https://github.com/user-attachments/assets/3ea1d4c4-eb30-4ade-a0c5-c6fdd6d7dce5)
![image](https://github.com/user-attachments/assets/8b61eac4-9958-42ae-b7e7-d6f54af42f50)
![image](https://github.com/user-attachments/assets/66bbdbd5-fb7d-4c02-82a1-4b5855872428)

**Reply by wesleyjcole** [reply]
Some pieces of this should be from degradation (e.g., the last plot you posted looks like it is from that).  The other outputs from Cambium are likely from curtailment.  We recently changed ReEDS to no longer allow rooftop PV to be curtailed, but I don't know if that has been implemented in PLEXOS for running the Cambium outputs.
